src/interfaces/vault.ts

```diff
@@ -451,38 +451,30 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     options: DepositOptions = {},
     overrides: CallOverrides = {}
   ): Promise<TransactionResponse> {
+    if (token === EthAddress) {
+      throw new SdkError("deposit:v2:eth not implemented");
+    }
```

Comment:
Users should be allowed to deposit eth into any vault, through a zap. The only difference is that it does not require approving, since its not an ERC20 token.

Reply:
Ahh my bad, moved the throwing after the zap checks with https://github.com/yearn/yearn-sdk/pull/280/commits/3e2c60b8d39ad9203608317ae452b67c56d2d97b